api-2.2/src/main/java/org/openmrs/module/fhir2/api/dao/impl/FhirConditionDaoImpl_2_2.java

```diff
@@ -42,6 +52,47 @@ public Condition getConditionByUuid(String uuid) {
 		        .uniqueResult();
 	}
 	
+	private ConditionClinicalStatus convertStatus(String status) {
+		if ("active".equalsIgnoreCase(status)) {
+			return ConditionClinicalStatus.ACTIVE;
+		}
+		if ("inactive".equalsIgnoreCase(status)) {
+			return ConditionClinicalStatus.INACTIVE;
+		}
+		// TODO during review: What value of the spec. should be mapped to HISTORY_OF?
+		// http://www.hl7.org/fhir/valueset-condition-clinical.html
+		return ConditionClinicalStatus.HISTORY_OF;
+	}
+	
+	@Override
+	public Collection<Condition> searchForConditions(ReferenceAndListParam patientParam, ReferenceAndListParam subjectParam,
+	        TokenOrListParam code, TokenOrListParam clinicalStatus, DateParam onsetDate, QuantityParam onsetAge,
+	        DateParam recordedData, SortSpec sort) {
+		Criteria criteria = sessionFactory.getCurrentSession().createCriteria(Condition.class);
+		
+		handlePatientReference(criteria, patientParam);
+		// TODO during review: Do we require a ":Patient" modifier or a "Patient/" type for the "subject" search param?
+		// And do we want to support anything other than "Patient" under "subject" based searches (e.g., "Group")?
+		handlePatientReference(criteria, subjectParam);
```

Comment:
The `patient` search parameter is defined as `Condition.subject.where(resolve() is Patient)`, i.e., if the `patientParam` is valid, we should not handle the `subjectParam` at all.

> Do we require a ":Patient" modifier or a "Patient/" type for the "subject" search param?

No, but we should support these where present

> Do we want to support anything other than "Patient" under "subject" based searches (e.g., "Group")?

We currently haven't implemented the concept of "Group" in the OpenMRS FHIR module, so we needn't support it here.